src/main/scala/org/zalando/spearheads/innkeeper/services/EskipRouteService.scala

```diff
@@ -1,28 +1,33 @@
 package org.zalando.spearheads.innkeeper.services
 
 import java.time.LocalDateTime
+
 import akka.NotUsed
 import akka.stream.scaladsl.Source
 import com.google.inject.Inject
 import org.zalando.spearheads.innkeeper.api.JsonProtocols._
-import org.zalando.spearheads.innkeeper.api.{EskipRoute, EskipRouteWrapper, NewRoute, RouteName}
+import org.zalando.spearheads.innkeeper.api.{EskipRoute, EskipRouteWrapper, NewRoute, RouteName, RouteOut}
 import org.zalando.spearheads.innkeeper.dao.{PathRow, RouteRow, RoutesRepo}
+import slick.backend.DatabasePublisher
 import spray.json.pimpString
 
 class EskipRouteService @Inject() (routesRepo: RoutesRepo, routeToEskipTransformer: RouteToEskipTransformer) {
 
   private val arrow = "\n -> "
 
-  def currentEskipRoutes(currentTime: LocalDateTime = LocalDateTime.now()): Source[EskipRouteWrapper, NotUsed] = {
+  def findModifiedSince(
+    since: LocalDateTime,
+    currentTime: LocalDateTime = LocalDateTime.now()): Source[EskipRouteWrapper, NotUsed] = {
 
-    Source.fromPublisher(routesRepo.selectLatestActiveRoutesWithPathPerName(currentTime).mapResult {
-      case (routeRow, pathRow) =>
-        EskipRouteWrapper(
-          RouteName(routeRow.name),
-          routeToEskipString(routeRow, pathRow),
-          routeRow.createdAt,
-          routeRow.deletedAt)
-    })
+    routeAndPathRowsStreamToEskipStream{
+      routesRepo.selectModifiedSince(since, currentTime)
+    }
+  }
+
+  def currentEskipRoutes(currentTime: LocalDateTime = LocalDateTime.now()): Source[EskipRouteWrapper, NotUsed] = {
+    routeAndPathRowsStreamToEskipStream{
+      routesRepo.selectActiveRoutesWithPath(currentTime)
+    }
```

Comment:
This looks a bit funky but I can't figure out a better way of writing this. I mean `routeAndPathRowsStreamToEskipStream`.